test/monitoring_console/monitoring_console_test.go

```diff
@@ -0,0 +1,532 @@
+			Expect(strings.Contains(peerList[0], podName)).To(Equal(true))
+
+			// Scale Standalone instance
+			testenvInstance.Log.Info("Scaling standalone cluster")
+			_, err = exec.Command("kubectl", "scale", "standalone", "-n", testenvInstance.GetName(), deployment.GetName(), "--replicas=2").Output()
+			Expect(err).To(Succeed(), "Failed to execute scale up command")
+
+			// Ensure standalone is scaling up
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), standalone)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for standalone status to be Scaling Up", "instance", standalone.ObjectMeta.Name, "Phase", standalone.Status.Phase)
+				testenv.DumpGetPods(testenvInstance.GetName())
+				return standalone.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseScalingUp))
+
+			// Ensure Standalone go to Ready phase
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), standalone)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for standalone status to be READY", "instance", standalone.ObjectMeta.Name, "Phase", standalone.Status.Phase)
+				testenv.DumpGetPods(testenvInstance.GetName())
+				return standalone.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), standalone)
+				return standalone.Status.Phase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Wait for new MC Pod to come up and existing MC Pod to terminate
+			Eventually(func() int {
+				count := testenv.GetMCPodCount(testenvInstance.GetName())
+				testenvInstance.Log.Info("Waiting for montioring console pods count", "count", count)
+				testenv.DumpGetPods(testenvInstance.GetName())
+				return count
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(1))
+
+			// Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Only 2 peer expected in MC peer list
+			peerList = testenv.GetConfiguredPeers(testenvInstance.GetName())
+			testenvInstance.Log.Info("Peers in configuredPeer List", "count", len(peerList))
+			Expect(len(peerList)).To(Equal(2))
+
+			// Check standalone pods are configured  in MC Peer List
+			found := make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 2; i++ {
+				podName := "splunk-" + deployment.GetName() + "-standalone-" + strconv.Itoa(i)
+				found[podName] = false
+				for _, peer := range peerList {
```

Comment:
A couple of things  - 
1. I don't think you need the outer for loop here. You anyways know from the check in line 243 that you have only 2 peers in the peer list. So you can keep a counter `i` in this loop itself, starting from 0 and increment at the end of the loop.
So something like this - 
```var podName string
found[podName] = false
i :=0
for _, peer := range peerList {
    podName = "splunk-" + deployment.GetName() + "-standalone-" + strconv.Itoa(i)
    if strings.Contains(peer, podName) {
	testenvInstance.Log.Info("Check Peer matches standalone pod", "Standalone Pod", podName, "Peer in peer list", peer)
	found[podName] = true
    }
    i++
}
```
2. You also don't need to `break` from the loop if you use the above logic.